### DISCUSSION TITLE
Folder Mapping - destination folder outside web root

### DISCUSSION THREAD

**Discussion by BlueuGIT** [Natizyskunk/vscode-sftp#130] [Q&A] [answered]
Hi - is there a way to map an individual folder to a remote folder outside the webroot folder (but still within the same /home directory structure?). The folder is accessed via a website alias.
Thanks

**Comment by BlueuGIT** [comment]
Nothing? My usual IDE lets me map on a folder-to-folder basis - I didn't realise that it was such a rare quality. VsCode looks absolutely brilliant, but this is a huge void.
Is it possible to use profiles or contexts, but have them all active at the same time, maybe automatically running on save using gulp?

**Reply by BlueuGIT** [reply]
I have just been told that my query may not be clear - to clarify:
I simply want to select an individual folder within my VSCode project and have it publish to a separate destination on my server - this destination is a folder which is outside of the /public_html structure, but still within the same /home folder structure.
I'd be very grateful for either advice on how to achieve this, or definitive confirmation that it is not possible. Many thanks.

**Comment by Natizyskunk** [comment, marked as answer]
I'm really not sure to understand your question. 

> "The destination is a folder outside of the /public_html".

What do you mean by that ?
The default destination folder is define by the `remotePath` option. Are you saying that your `remotePath` is set to `/public_html` ?

> "But still within the same /home folder structure". 

Which `/home` folder are you talking about ? The one at the root of your remote wher all your users are ?

If I understand correctly you have currently a project that has the `remotePath` option set to `/public_html`. `/public_html` being a folder at the root level of your remote.
You want to be able to select a file inside your local project and upload it to your remote to another destination than the one defined in your `remotePath` option.

Am I right ?

If this is what you xant you should be able to achieve it using multiple contexts like this : 
```json
[
    {
        "name": "server1",
        "host": "<server1-host>",
        "context": ".",  // <-- A path relative to the workspace root folder (local project root).
        "username": "username",
        "password": "password",
        "remotePath": "/public_html"
        "ignore": ["folder1"],
    },
    {
        "name": "server1-bis",
        "host": "<server1-host>",
        "context": "./folder1", // <-- A path relative to the workspace root folder (local project root).
        "username": "username",
        "password": "password",
        "remotePath": "/home/folder1"
    }
]
```
Now for the fisrt context, all files/folders in your local projet (project root folder) will be uploaded to the "/plublic_htoml" folder on the remote except the "folder 1" folder.
For the second context, only files/folders inside the "folder1" folder will be uploaded  to the "/home/folder" folder on the remote.

Note : All contexts are runned at the same time.
